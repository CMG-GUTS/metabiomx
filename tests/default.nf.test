nextflow_pipeline {

    name "TEST DECONTAMINATION SUBWORKFLOW"
    script "../main.nf"
    tag "decontamination"
    tag "pipeline"

    test("TEST DECONTAMINATION SUBWORKFLOW") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.trace.succeeded().size()
                ).match() },
                { assert snapshot(
                    path("${outputDir}/decontamination")
                    .list()
                    .collect { getRecursiveFileNames(it, outputDir) }
                    .flatten()
                ).match("decontamination") },
                                { assert snapshot(
                    path("${outputDir}/multiqc")
                    .list()
                    .collect { getRecursiveFileNames(it, outputDir) }
                    .flatten()
                ).match("multiqc") }
            )
        }
    }
}

def getRecursiveFileNames(fileOrDir, outputDir) {
    if(file(fileOrDir.toString()).isDirectory()) {
        return fileOrDir.list().collect { getRecursiveFileNames(it, outputDir) }
    }
    return fileOrDir.toString().replace("${outputDir}/", "")
}