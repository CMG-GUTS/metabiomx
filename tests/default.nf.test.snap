{
    "decontamination": {
        "content": [
            [
                "decontamination/S0001_paired_trim_1_kneaddata_paired_1.fastq.gz",
                "decontamination/S0001_paired_trim_1_kneaddata_paired_2.fastq.gz"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-13T11:07:29.287111"
    },
    "multiqc": {
        "content": [
            [
                "multiqc/decon_multiqc_report.html",
                "multiqc/merged_multiqc_stats.tsv",
                "multiqc/raw_multiqc_report.html",
                "multiqc/trimmed_multiqc_report.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-13T11:07:29.325428"
    },
    "TEST DECONTAMINATION SUBWORKFLOW": {
        "content": [
            9
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-12T17:45:50.795891"
    }
}